portfolio/src/main/webapp/script.js

```diff
@@ -41,3 +41,15 @@ function addRandomFunFact() {
   // Add it to the page. 
   funFactContainer.innerText = newFact;
 }
+
+/**
+ * Displays a greeting on the page using fetch
+ */
+ async function displayGreeting() {
+   // get greeting from /data page
+   const response = await fetch('/data');
+   const greeting = await response.text();
+
+   // display greeting in greeting container
+   document.getElementById('greeting-container').innerHTML = greeting;
```

Comment:
I would recommend using .innerText instead of innerHTML if you're only intending to use it for text content. Adding HTML like this can lead to all sorts of weird bugs (and potentially security vulnerabilities) if you're not careful.

Reply:
Thank you! That's very good to know, fix has been pushed.